server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -946,7 +951,7 @@ private void checkForTimeout(long requestId) {
         assert responseHandlers.contains(requestId) == false;
         TimeoutInfoHolder timeoutInfoHolder = timeoutInfoHandlers.remove(requestId);
         if (timeoutInfoHolder != null) {
-            long time = System.currentTimeMillis();
+            long time = threadPool.relativeTimeInMillis();
```

Comment:
So this is a good change, not even for the purposes of testing, but in general from an absolute clock to a relative clock for measuring elapsed intervals. Is the idea here that in tests we will mock `ThreadPool#relativeTimeInMillis`?

Reply:
Yes - in fact we [mock out most of `ThreadPool`](https://github.com/elastic/elasticsearch/blob/02b483c372683a0abfde2f7c398463e58d181581/test/framework/src/main/java/org/elasticsearch/cluster/coordination/DeterministicTaskQueue.java#L273) so we can write deterministic tests that emulate concurrent activity (across multiple nodes) and timeouts without needing to wait in realtime for timeouts to occur. This particular change only affects the logged messages; because we don't wait in realtime without this change the reported timeouts were confusingly short.